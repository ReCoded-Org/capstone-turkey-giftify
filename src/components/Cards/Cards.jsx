import React from "react";
import SingleCard from "../SingleCard/SingleCard";

function Cards({ category, cardsBySearch, singleCard, itemsDetails}) {
  return (
    <div>
    {singleCard === true ? <SingleCard itemsDetails={itemsDetails}/> : 
        <div>  {cardsBySearch} {category} </div>     
    }
    </div>
  );
}

export default Cards;
