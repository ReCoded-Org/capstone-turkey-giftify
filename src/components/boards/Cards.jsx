import React from "react";
import SingleCard from "../SingleCard/SingleCard";
import CardsBySearch from "../boards/CardsBySearch";
import Categories from "../boards/Categories";
import { BrowserRouter as Router, Route } from "react-router-dom";

function Cards({
  allProducts,
  allCategories,
  singleCard,
  setSingleCard,
  itemsDetails,
  setItemsDetails,
  selectedItems,
  setSelectedItems,
  testSearchValue,
}) {
  return (
    <div>
      <Router>
        <Route exact path="/">
          <CardsBySearch
            allProducts={allProducts}
            setSingleCard={setSingleCard}
            setItemsDetails={setItemsDetails}
            testSearchValue={testSearchValue}
          />

          <Categories
            allProducts={allProducts}
            allCategories={allCategories}
            setSelectedItems={setSelectedItems}
            selectedItems={selectedItems}
            setSingleCard={setSingleCard}
            setItemsDetails={setItemsDetails}
          />
        </Route>

        <Route path="/productDetails">
          <SingleCard itemsDetails={itemsDetails} />
        </Route>
      </Router>
    </div>
  );
}

export default Cards;
