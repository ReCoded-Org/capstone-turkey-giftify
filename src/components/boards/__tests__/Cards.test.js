import React from "react";
import Cards from "../Cards";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("Cards rendered correctly", () => {
  const tree = renderer.create(<Cards allProducts />).toJSON();
  expect(tree).toMatchSnapshot();
});
