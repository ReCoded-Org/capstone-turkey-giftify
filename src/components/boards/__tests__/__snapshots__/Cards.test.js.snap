// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards rendered correctly 1`] = `
<div>
  <div
    className="container"
  >
    <div
      className="row cards"
    />
  </div>
  <div
    className="categories"
  >
    <button
      className="oneCategory book"
      onClick={[Function]}
    >
      book
    </button>
  </div>
  <div
    className="categories"
  >
    <button
      className="oneCategory clothing"
      onClick={[Function]}
    >
      clothing
    </button>
  </div>
</div>
`;
